quadriga/src/main/java/edu/asu/diging/quadriga/web/DisplayCollectionController.java

```diff
@@ -0,0 +1,95 @@
+
+import edu.asu.diging.quadriga.core.exceptions.CollectionNotFoundException;
+import edu.asu.diging.quadriga.core.exceptions.InvalidObjectIdException;
+import edu.asu.diging.quadriga.core.model.Collection;
+import edu.asu.diging.quadriga.core.model.EventGraph;
+import edu.asu.diging.quadriga.core.model.MappedTripleGroup;
+import edu.asu.diging.quadriga.core.service.CollectionManager;
+import edu.asu.diging.quadriga.core.service.EventGraphService;
+import edu.asu.diging.quadriga.core.service.MappedTripleGroupService;
+
+@Controller
+public class DisplayCollectionController {
+
+    @Autowired
+    private CollectionManager collectionManager;
+
+    @Autowired
+    private EventGraphService eventGraphService;
+    
+    @Autowired
+    private MappedTripleGroupService mappedTripleGroupService;
+    
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
+    @RequestMapping(value = "/auth/collections/{id}", method = RequestMethod.GET)
+    public String get(@PathVariable String id, Model model) {
+        
+        Collection collection;
+        try {
+            collection = collectionManager.findCollection(id);
+            if(collection == null) {
+                return "error404Page";
+            }
+        } catch (InvalidObjectIdException e) {
+            return "error404Page";
+        }
+        
+        ObjectId collectionId = collection.getId();
+        
+        List<EventGraph> eventGraphs = eventGraphService.findAllEventGraphsByCollectionId(collectionId);
+        
+        if(!eventGraphs.isEmpty()) {
+            // We get last network submission info by getting the last EventGraph which will be a part of the last network
+            EventGraph lastNetwork = eventGraphs.get(0);
+            model.addAttribute("lastNetworkSubmittedAt", lastNetwork.getCreationTime());
+            model.addAttribute("lastNetworkSubmittedBy", lastNetwork.getAppName());
+        }
+        
+        model.addAttribute("collection", collection);
+        
+        // One network may have multiple eventGraphs, but all of them will have same sourceURI in the context
+        // This sourceURI will be used to group eventGraphs together that belong to the same network
+        model.addAttribute("numberOfSubmittedNetworks", eventGraphs.stream().collect(Collectors.groupingBy(event -> event.getContext().getSourceUri())).size());
+        
+        // Get default mappings from Concepts
+        model.addAttribute("defaultMappings", getDefaultMappings(collectionId.toString()));
+        
+        return "auth/displayCollection";
+        
+    }
+    
+    /**
+     * This method gets all mappedTripleGroups that have the current collectionId for which
+     * the mappedTripleType is "DefaultMapping"
+     * 
+     * @param collectionId used to find mappedTripleGroupId
+     * @return the number of default mappings
+     */
+    private int getDefaultMappings(String collectionId) {
```

Comment:
I'm confused, what does this method do? It's called getDefaultMappings but it returns an int?

Reply:
It returns number of default mappings
I'll update the method name to getNumberOfDefaultMappings

**This is as per our discussion on slack:**

Julia Lessios-Damerow  11:56 AM
The table would list all the mappings (right now only one, Default Mapping) and a number e.g. 10. Once we add functionality to add more mappings, the list would have eg Default Mappings (10), WiC Mappings (7), Test Mappings (12) etc

Reply:
I think this is an issue of confusing namings. A method named `getDefaultMappings` is expected to return default mappings. This method returns the number of triples in the default mapping group though, right?

Reply:
Yes, I have updated method name to getNumberOfDefaultMappings